.xe/features/engineering-context/spec.md

```diff
@@ -0,0 +1,99 @@
+dependencies: []
+---
+
+<!-- markdownlint-disable single-title -->
+
+# Feature: Engineering Context Templates
+
+## Problem
+
+Architects and engineers need a structured way to document technical architecture, engineering principles, and development processes for AI-powered development. Without token-efficient templates, engineering context documentation contains unnecessary verbosity, leading to excessive AI token consumption while reducing clarity.
+
+## Goals
+
+- Provide token-efficient templates for essential architecture/engineering-owned context (architecture, engineering principles, development process)
+- Enable AI agents to make technical implementation decisions aligned with architecture patterns and engineering standards
+- Minimize token usage while maintaining clarity and decision-making utility
+
+Explicit non-goals:
+
+- This feature does NOT include standards files (`.xe/standards/` - separate concern)
+- This feature does NOT include playbook templates (separate feature)
+
+## Scenario
+
+- As an **Architect**, I need a concise template for documenting technology stack and architecture patterns so that AI can make aligned technical decisions
+  - Outcome: Token-efficient architecture documentation guiding all technical implementation
+
+- As an **Engineer**, I need a template for engineering principles that AI can consume without excessive token overhead so that implementation follows consistent standards
+  - Outcome: Clear engineering principles without verbose instructions or duplicate content
+
+- As an **AI Agent**, I need to read architecture and engineering context during feature implementation so that I can align with technical decisions and quality standards
+  - Outcome: AI-generated code aligns with architecture patterns and engineering principles
+
+## Success Criteria
+
+- Architecture.md template provides comprehensive technical architecture guidance while maintaining token efficiency
+- Engineering.md template provides actionable engineering principles and technical standards
+- Templates follow [template standard](.xe/standards/catalyst.md)
+- Templates enable AI agents to make technical decisions without reading unnecessary context
+
+## Design principles
+
+See [.xe/product.md](.xe/product.md) for product-wide design principles.
+
+No feature-specific design principles are needed for this template feature.
+
+## Requirements
+
+### Functional Requirements
+
+- **FR-1**: Architecture.md template MUST exist
+  - **FR-1.1**: Template MUST follow template standard defined in `.xe/standards/catalyst.md`
+  - **FR-1.2**: Template MUST include Overview section with pointers to related context files
+  - **FR-1.3**: Template MUST include Technology Stack section defining runtime environment, data storage, automation tools, AI integration, testing framework, deployment method, security measures, and monitoring/logging
```

Comment:
Should each of these be a separate FR? I'm fine either way. Just asking.

Reply:
Yes, I like that idea. Please do.

But in this case, do we want to prescribe the areas that need coverage? If we don't, every time we update the spec it could result in a different determination of what is ideal, since the spec isn't specific enough. What you have here isn't all-inclusive and will likely lead to different results on consecutive runs. If we're precise, we can guarantee the output always has what we intend; nothing more, nothing less.

Reply:
I'll repeat my comment: Do we want to prescribe the areas that need coverage? If we don't, every time we update the spec it could result in a different determination of what is ideal, since the spec isn't specific enough. What you have here isn't all-inclusive and will likely lead to different results on consecutive runs. If we're precise, we can guarantee the output always has what we intend; nothing more, nothing less.